docs/DISCOVERABLE_PARTITIONS.md

```diff
@@ -0,0 +1,359 @@
+---
+title: Discoverable Partitions Specification
+category: Concepts
+layout: default
+---
+# The Discoverable Partitions Specification
+
+_TL;DR: Let's automatically discover, mount and enable the root partition,
+`/home/`, `/srv/`, `/var/` and `/var/tmp/` and the swap partitions based on
+GUID Partition Tables (GPT)!_
```

Comment:
I'm not reviewing this commit, since it's just an import of existing docs. Any fixes (stylistic or otherwise) can go in later. It seems that github renders this reasonably well.